cabal-install/Distribution/Client/TargetSelector.hs

```diff
@@ -479,6 +486,8 @@ resolveTargetSelector ppinfo opinfo targetStrStatus =
       Unambiguous target -> Right target
 
       None errs
+        | TargetStringFileStatus1 str _ <- targetStrStatus
+        , validPackageName str -> Right (TargetPackageName (mkPackageName str))
```

Comment:
In my tests we only fall here when i run
`cabal new-build some-package-not-in-the-project`
while if I run
`cabal new-build some-extra-package`
we fall in the `Unambiguous target` case.

I don't understand if
1. it should match the extra package
2. it should match any other package (ex. from hackage)

If (1): this doesn't work and we probably need to re-add `syntaxForm1Name` and filter out extra packages in `syntaxForm1Package`.

If (2): Well, until now i thought we were dealing with the first case... I'll have to think abou this.